src/genericsite/templates/genericsite/base.html

```diff
@@ -17,21 +17,49 @@
   crossorigin="anonymous">
   {% endblock bootstrap_styles %}
   {% block opengraph %}{{ opengraph }}{% endblock %}
-  {% block extra_head %}{% endblock %}
+  {% block extra_head %}
+  <link rel="stylesheet" href="{% static 'generic.css' %}">
+  {% endblock %}
 </head>
 <body class="{% block body_class %}{% endblock body_class %}">
-  {% block header %}{% include header_template|default:"genericsite/blocks/header_simple.html" %}{% endblock header %}
-  {% block precontent %}{% include precontent_template|default:"genericsite/blocks/empty.html" %}{% endblock precontent %}
-  {% block content %}{% include content_template|default:"genericsite/blocks/article_text.html" %}{% endblock content %}
-  {% block postcontent %}{% include postcontent_template|default:"genericsite/blocks/empty.html" %}{% endblock postcontent %}
-  {% block footer %}{% include footer_template|default:"genericsite/blocks/footer_simple.html" %}{% endblock footer %}
+  {% block header %}
+  <header class="{{ bootstrap_container_class }}">
+    {% include header_template|default:"genericsite/blocks/header_simple.html" %}
+  </header>
+  {% endblock header %}
+
+  {% block precontent %}
+  <div class="precontent {{ bootstrap_container_class }}">
+    {% include precontent_template|default:"genericsite/blocks/empty.html" %}
+  </div>
+  {% endblock precontent %}
+
+  {% block content %}
+  <main role="main" class="{{ bootstrap_container_class }}">
+    {% include content_template|default:"genericsite/blocks/article_text.html" %}
+  </main>
+  {% endblock content %}
+
+  {% block postcontent %}
+  <div class="postcontent {{ bootstrap_container_class }}">
+    {% include postcontent_template|default:"genericsite/blocks/empty.html" %}
+  </div>
+  {% endblock postcontent %}
+
+  {% block footer %}
+  <footer class="{{ bootstrap_container_class }}">
+    {% include footer_template|default:"genericsite/blocks/footer_simple.html" %}
+  </footer>
+  {% endblock footer %}
+
   {% block bootstrap_js %}
```

Comment:
Is there a reason this isn't in the `<head>`?

Reply:
This is an old-school performance hack. Scripts block the page render, so placing scripts that don't affect layout at the end before the close body tag improves perceived performance by allowing the browser to render the rest of the page before downloading and parsing the scripts. In modern browsers, one could also place these scripts in the head and add the `defer` or `async` attribute to get the same effect. Old habits die hard.